### PR TITLE
[sail] Update to 0.9.0-pre14

### PR DESCRIPTION
**Pull request by HappySeaFox**
**Describe the pull request**

Numerous improvements in upstream.

- #### What does your PR fix?  
\-

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
Same as before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/sail/portfile.cmake [hunk 1]
```diff
@@ -29,33 +29,20 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
...
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/share/${PORT}manip)
-file(GLOB SAIL_MANIP_CONFIGS "${CURRENT_PACKAGES_DIR}/share/${PORT}/SailManip*")
-foreach(SAIL_MANIP_CONFIG IN LISTS SAIL_MANIP_CONFIGS)
-    get_filename_component(SAIL_MANIP_CONFIG_NAME "${SAIL_MANIP_CONFIG}" NAME)
-    file(RENAME ${SAIL_MANIP_CONFIG} ${CURRENT_PACKAGES_DIR}/share/${PORT}manip/${SAIL_MANIP_CONFIG_NAME})
-endforeach()
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
 
 # Handle copyright
 file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/sail/portfile.cmake:48]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/sail/portfile.cmake [hunk 2]
```diff
@@ -3,24 +3,23 @@ vcpkg_fail_port_install(ON_TARGET "UWP")
...
+    REF v0.9.0-pre14
+    SHA512 a1b50893a9d4112f2bf2ca73e763d705615d0b110a41a1dbe8c32566b7706c054cb44d9c169163ef4f9d125242b39b55e123021fb2dad8d2f5a4d216c1a202b4
     HEAD_REF master
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SAIL_STATIC)
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/sail/portfile.cmake:14]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"

```

File: ports/sail/portfile.cmake [hunk 3]
```diff
@@ -29,33 +28,20 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
...
 # Move cmake configs
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/sail)
+vcpkg_cmake_config_fixup(PACKAGE_NAME sail       CONFIG_PATH lib/cmake/sail       DO_NOT_DELETE_PARENT_CONFIG_PATH)
+vcpkg_cmake_config_fixup(PACKAGE_NAME sailcodecs CONFIG_PATH lib/cmake/sailcodecs DO_NOT_DELETE_PARENT_CONFIG_PATH)
+vcpkg_cmake_config_fixup(PACKAGE_NAME sailcommon CONFIG_PATH lib/cmake/sailcommon DO_NOT_DELETE_PARENT_CONFIG_PATH)
+vcpkg_cmake_config_fixup(PACKAGE_NAME sailc++    CONFIG_PATH lib/cmake/sailc++    DO_NOT_DELETE_PARENT_CONFIG_PATH)
+vcpkg_cmake_config_fixup(PACKAGE_NAME sailmanip  CONFIG_PATH lib/cmake/sailmanip  DO_NOT_DELETE_PARENT_CONFIG_PATH)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib/cmake
+                    ${CURRENT_PACKAGES_DIR}/debug/lib/cmake)
```
Comment (JonLiu1993) [inline ports/sail/portfile.cmake:38]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake"
                    "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")

```

File: ports/sail/portfile.cmake [hunk 4]
```diff
@@ -29,33 +28,20 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include
...
-    file(RENAME ${SAIL_CPP_CONFIG} ${CURRENT_PACKAGES_DIR}/share/${PORT}c++/${SAIL_CPP_CONFIG_NAME})
-endforeach()
-
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/share/${PORT}manip)
-file(GLOB SAIL_MANIP_CONFIGS "${CURRENT_PACKAGES_DIR}/share/${PORT}/SailManip*")
-foreach(SAIL_MANIP_CONFIG IN LISTS SAIL_MANIP_CONFIGS)
-    get_filename_component(SAIL_MANIP_CONFIG_NAME "${SAIL_MANIP_CONFIG}" NAME)
-    file(RENAME ${SAIL_MANIP_CONFIG} ${CURRENT_PACKAGES_DIR}/share/${PORT}manip/${SAIL_MANIP_CONFIG_NAME})
-endforeach()
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```
Comment (JonLiu1993) [inline ports/sail/portfile.cmake:44]:
```suggestion
file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@HappySeaFox ,When we update in the future, the path in the portfile.cmake file must be added with ""

File: ports/sail/vcpkg.json [hunk 5]
```diff
@@ -1,13 +1,22 @@
 {
   "name": "sail",
-  "version-semver": "0.9.0-pre13",
+  "version-semver": "0.9.0-pre14",
   "description": "The missing small and fast image decoding library for humans (not for machines)",
   "homepage": "https://github.com/smoked-herring/sail",
   "supports": "!uwp",
   "dependencies": [
     "giflib",
+    "libavif",
```
Comment (JonLiu1993) [inline ports/sail/vcpkg.json:9]: @HappySeaFox ,Does the "gdal" port have any special dependencies in the new version? I checked the source and it seems that there is no special change from the previous version in terms of "gdal"

Comment (HappySeaFox) [reply ports/sail/vcpkg.json:9]: What's gdal? 😀

Comment (JonLiu1993) [reply ports/sail/vcpkg.json:9]: “libavif”,sorry, I made a mistake

Comment (HappySeaFox) [reply ports/sail/vcpkg.json:9]: pre14 added support of avif format implemented on top of libavif. See https://github.com/HappySeaFox/sail/blob/v0.9.0-pre14/src/sail-codecs/avif/avif.cmake . So libavif is a direct dependency.

Comment (JonLiu1993) [reply ports/sail/vcpkg.json:9]: Thanks for your clarification.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**